### ISSUE TITLE
Usage for static barcoding data?

### ISSUE THREAD

**Issue by cnk113** [shadowdeng1994/TarCA.beta#1]
Hi,

I was wondering how you would format the input for static barcoding datasets?

Best,
Chang

**Comment by shadowdeng1994** [comment]
Dear Chang, 

Thank you for your interest in our research. 
We did apply TarCA on static barcodes and you could find more details in Supplementary Note 4 in our manuscript. 
In short, given a cell population, those sharing the same static bacodes would be treated as a monophyletic clade. Then we could obtain the distribution of clade size directly and calculate the coalescent rate to infer Np. 

Sincerely,
Shanjun

**Comment by cnk113** [comment]
Hi Shanjun,

Thanks for the clarification, could you perhaps provide an example input of the static data?
It's kind of hard to follow the documentation, I'm assuming I have addMono clades but I want to know the specific format.

Best,
Chang

**Comment by shadowdeng1994** [comment]
> Hi Shanjun,
> 
> Thanks for the clarification, could you perhaps provide an example input of the static data? It's kind of hard to follow the documentation, I'm assuming I have addMono clades but I want to know the specific format.
> 
> Best, Chang

Dear Chang,

Unfortunately, there's no ready-to-use function in the previous version. But, we're glad to say that we have updated an additional function for handling static barcodes in the lastest version now. You can either reinstall the lastest package or copy the following script directly. 

```
Np_Est_StaticBC <- function(BCTable,Ann){
    message("===> Estimating Np.")
    
    tmp.CladeSizeTable <- 
    Ann %>% 
    left_join(BCTable) %>% 
    group_by(TipAnn,StaticBC) %>% summarise(CladeSize=n()) %>% 
    group_by(TipAnn,CladeSize) %>% summarise(Freq=n()) %>% 
    arrange(TipAnn,CladeSize)
    
    tmp.res <- 
    tmp.CladeSizeTable %>% 
    group_by %>% dplyr::mutate(LLL=paste0(CladeSize," (",Freq,")")) %>% 
    group_by(TipAnn) %>% summarise(MonoInfo=paste(LLL,collapse = ", "),Total=sum(CladeSize*Freq),Np=Total*(Total-1)/sum(CladeSize*(CladeSize-1)*Freq))
    
    return(tmp.res)
}
```

There're two inputs required. One dataframe contains the information of the static barcode with two column (TipLabel and StaticBC) and the other one is the annotation for each cells. 
> | TipLabel | StaticBC |
> | --- | --- |
> | Cell_1 | BC1 |
> | Cell_2 | BC1 |
> | Cell_3 | BC2 |
> | Cell_4 | BC3 |
> | Cell_5 | BC2 |
> | Cell_6 | BC2 |
> | Cell_7 | BC1 |
> | Cell_8 | BC3 |
> | Cell_9 | BC3 |
> | Cell_10 | BC2 |

BTW, I should note that, since static barcoding system is a kind of lineage tracing system with relatively low resolution, the inferred Np may largely rely on the uniformity and diversity of the barcoding system that you used. Therefore, I think the relative value between samples might be more informative than the absolute value of estimated Np in such situation. 

Let me know if you need any further assistance.

Best wish,
Shanjun

**Comment by cnk113** [comment]
Hi Shanjun I had another quick question on how to get the plotting of tree with the output of the static barcoding dataset?

Best,
Chang

**Comment by shadowdeng1994** [comment]
> Hi Shanjun I had another quick question on how to get the plotting of tree with the output of the static barcoding dataset?
> 
> Best, Chang

Hi Chang,

In my point of view, it's hard to reconstruct a cell phylogeny with static barcodes as it typically relies on a predefined but unique barcode to trace a progenitor and its descendants.  Once barcoding is complete, no further information about subsequent cell division events can be appended to the barcode anymore. 

Therefore, if you want to obtain a cell phylogeny with sequential information of each cell division, I think an evolving barcoding system with high resolution (e.g. [SMALT](https://www.nature.com/articles/s41592-021-01325-x)) would be more suitable for your needs. 

Best wish,
Shanjun
